integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java

```diff
@@ -0,0 +1,142 @@
+   */
+  public static Callable<Boolean> checkDomainEvent(
+      String opNamespace, String domainNamespace, String domainUid, String reason, String type, DateTime timestamp) {
+    return () -> {
+      logger.info("Verifying {0} event is logged by the operator in domain namespace {1}", reason, domainNamespace);
+      try {
+        List<V1Event> events = Kubernetes.listNamespacedEvents(domainNamespace);
+        for (V1Event event : events) {
+          if (event.getReason().contains(reason)
+              && event.getMetadata().getCreationTimestamp().isAfter(timestamp.getMillis())) {
+            logger.info(Yaml.dump(event));
+            verifyOperatorDetails(event, opNamespace, domainUid);
+            //verify reason
+            logger.info("Verifying domain event {0}", reason);
+            assertTrue(event.getReason().equals(reason));
+            //verify type
+            logger.info("Verifying domain event type {0}", type);
+            assertTrue(event.getType().equals(type));
+            return true;
+          }
+        }
+      } catch (ApiException ex) {
+        Logger.getLogger(ItKubernetesEvents.class.getName()).log(Level.SEVERE, null, ex);
+      }
+      return false;
+    };
+  }
+
+  // Verify the operator instance details are correct
+  private static void verifyOperatorDetails(V1Event event, String opNamespace, String domainUid) throws ApiException {
+    logger.info("Verifying operator details");
+    String operatorPodName = TestActions.getOperatorPodName(OPERATOR_RELEASE_NAME, opNamespace);
+    //verify DOMAIN_API_VERSION
+    assertTrue(event.getInvolvedObject().getApiVersion().equals(TestConstants.DOMAIN_API_VERSION),
+        "Expected " + TestConstants.DOMAIN_API_VERSION + " ,Got " + event.getInvolvedObject().getApiVersion());
+    //verify reporting component to be operator release
+    assertTrue(event.getReportingComponent().equals("weblogic.operator"),
+        "Didn't get reporting component as " + "weblogic.operator");
+    //verify reporting instance to be operator instance
+    assertTrue(event.getReportingInstance().equals(operatorPodName),
+        "Didn't get reporting instance as " + operatorPodName);
+    //verify the event was created by operator
+    Map<String, String> labels = event.getMetadata().getLabels();
+    assertTrue(labels.containsKey("weblogic.createdByOperator")
+        && labels.get("weblogic.createdByOperator").equals("true"));
+    //verify the domainUID matches
+    assertTrue(labels.containsKey("weblogic.domainUID")
+        && labels.get("weblogic.domainUID").equals(domainUid));
+  }
+
+  /**
+   * Returns the description for different event types.
+   *
+   * @param reason event to check for Created, Changed, deleted, processing etc
+   * @param domainUid UID of the domain
+   * @return String containing the complete description
+   */
+  public static String getDomainEventMessage(String reason, String domainUid) {
```

Comment:
now that we don't check event messages anymore, do we still need this method?

Reply:
removed